--- boost_1_35_0/boost/archive/polymorphic_iarchive.hpp.orig	2007-11-25 19:07:19.000000000 +0100
+++ boost_1_35_0/boost/archive/polymorphic_iarchive.hpp	2008-04-29 11:20:01.000000000 +0200
@@ -16,6 +16,7 @@
 
 //  See http://www.boost.org for updates, documentation, and revision history.
 
+#include <climits> // ULONG_MAX
 #include <cstddef> // std::size_t
 #include <boost/config.hpp>
 
--- boost_1_35_0/boost/archive/polymorphic_oarchive.hpp.orig	2007-11-25 19:07:19.000000000 +0100
+++ boost_1_35_0/boost/archive/polymorphic_oarchive.hpp	2008-04-29 11:20:17.000000000 +0200
@@ -16,6 +16,7 @@
 
 //  See http://www.boost.org for updates, documentation, and revision history.
 
+#include <climits> // ULONG_MAX
 #include <cstddef> // size_t
 #include <string>
 
