--- boost_1_46_1/tools/build/v2/build/configure.jam.orig	2010-11-17 00:02:36.000000000 +0100
+++ boost_1_46_1/tools/build/v2/build/configure.jam	2011-03-22 19:00:43.809098710 +0100
@@ -142,7 +142,7 @@
         {           
             local x = [ PAD "    - $(what)" : $(.width) ] ;            
             if [ UPDATE_NOW $(jam-targets) :
-                 $(.log-fd) : ignore-minus-n : ignore-minus-q ] 
+                 $(.log-fd) : ignore-minus-n ] 
             {
                 .$(what)-supported.$(ps) = yes ;
                 result = true ;
