--- boost_1_67_0/tools/build/src/tools/gcc.jam.orig	2018-04-11 15:49:09.000000000 +0200
+++ boost_1_67_0/tools/build/src/tools/gcc.jam	2018-07-16 19:40:36.332045183 +0200
@@ -356,11 +356,6 @@
 
     compile-link-flags <target-os>hpux/<address-model>32 : -milp32 ;
     compile-link-flags <target-os>hpux/<address-model>64 : -mlp64 ;
-
-    local generic-os = [ set.difference $(all-os) : aix hpux ] ;
-    local arch = power sparc x86 ;
-    compile-link-flags <target-os>$(generic-os)/<architecture>$(arch)/<address-model>32 : -m32 ;
-    compile-link-flags <target-os>$(generic-os)/<architecture>$(arch)/<address-model>64 : -m64 ;
 }
 
 {
@@ -624,11 +619,11 @@
 ###
 
 # Declare flags and action for compilation.
-toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
-toolset.flags gcc.compile OPTIONS <optimization>speed : -O3 ;
-toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
-toolset.flags gcc.compile OPTIONS <optimization>minimal : -O1 ;
-toolset.flags gcc.compile OPTIONS <optimization>debug : -Og ;
+toolset.flags gcc.compile OPTIONS <optimization>off   : ;
+toolset.flags gcc.compile OPTIONS <optimization>speed : ;
+toolset.flags gcc.compile OPTIONS <optimization>space : ;
+toolset.flags gcc.compile OPTIONS <optimization>minimal : ;
+toolset.flags gcc.compile OPTIONS <optimization>debug : ;
 
 toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
 toolset.flags gcc.compile OPTIONS <inlining>on   : -Wno-inline ;
@@ -637,8 +632,8 @@
 toolset.flags gcc.compile OPTIONS <warnings>pedantic : -Wall -Wextra -pedantic ;
 toolset.flags gcc.compile OPTIONS <warnings-as-errors>on : -Werror ;
 
-toolset.flags gcc.compile OPTIONS <debug-symbols>on : -g ;
-toolset.flags gcc.compile OPTIONS <profiling>on : -pg ;
+toolset.flags gcc.compile OPTIONS <debug-symbols>on : ;
+toolset.flags gcc.compile OPTIONS <profiling>on : ;
 
 toolset.flags gcc.compile OPTIONS <local-visibility>hidden : -fvisibility=hidden ;
 toolset.flags gcc.compile.c++ OPTIONS <local-visibility>hidden : -fvisibility-inlines-hidden ;
@@ -1069,7 +1064,7 @@
 # x86 and compatible
 # The 'native' option appeared in gcc 4.2 so we cannot safely use it as default.
 # Use i686 instead for 32-bit.
-toolset.flags gcc OPTIONS <architecture>x86/<address-model>32/<instruction-set> : -march=i686 ;
+toolset.flags gcc OPTIONS <architecture>x86/<address-model>32/<instruction-set> : ;
 cpu-flags gcc OPTIONS : x86 : native : -march=native ;
 cpu-flags gcc OPTIONS : x86 : i486 : -march=i486 ;
 cpu-flags gcc OPTIONS : x86 : i586 : -march=i586 ;
