--- boost_1_63_0/libs/python/build/Jamfile.orig	2016-12-22 13:33:19.000000000 +0100
+++ boost_1_63_0/libs/python/build/Jamfile	2017-04-08 21:35:49.226719994 +0200
@@ -140,7 +140,7 @@
             [ cond [ python.numpy ] : <library>/python//python_for_extensions ]
             [ unless [ python.numpy ] : <build>no ]
 	    <include>$(numpy-include)
-	    <library>boost_python
+	    [ cond $(is-py3) : <library>boost_python3 : <library>boost_python ]
             <python-debugging>on:<define>BOOST_DEBUG_PYTHON
             [ cond $(is-py3) : <python>$(py3-version) ]
 
