--- boost_1_31_0/libs/python/build/Jamfile.orig	2003-11-04 19:30:37.000000000 +0100
+++ boost_1_31_0/libs/python/build/Jamfile	2004-02-19 00:07:10.072772224 +0100
@@ -58,6 +58,7 @@
     : ../src/$(sources)
     : $(BOOST_PYTHON_V2_PROPERTIES)
       <define>BOOST_PYTHON_SOURCE
+      <find-library>python$(PYTHON_VERSION)
       $(bpl-linkflags)
         <msvc-stlport><release>$(msvc-stlport-workarounds)
         <darwin><*><linkflags>-bind_at_load
